src/components/search-form.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+
+import UserInfo from '../stores/user-info';
+import UserAction from '../actions/user-action';
+
+export default class SearchForm extends React.Component {
+    constructor (props) {
+        super(props);
+
+        this.state = {
```

Comment:
we should handle properly some edge case situations:
1. I either should be unable to add a new user during searching or adding should not affect on searching results
1. Saving changes after editing user info should not switch off searching mode. If after editing user no more match searching criteria than it should be excluded from the searching results
1. There is no user in the searching results. We should show some info message about it.